horace_core/sqw/file_io/@horace_binfile_interface/private/convert_old_det_forms_.m

```diff
@@ -0,0 +1,44 @@
+function det = convert_old_det_forms_(detpar,n_instances)
+% Method used to convert detector formats written by any previous version
+% of Horace into Horace 4.01 data form.
+%
+% Input:
+% detpar      -- an old or new format detector information.
+%                Normally obtained from binary sqw file.
+% n_instances -- number of run, this
+%
+% Returns:
+% det         -- detector information packed in
+%                unique_object_container container and
+%                distributed into approriate number of input
+%                runs.
+
+if isstruct(detpar) % we do not need do deal with struct array
+    % here as this have never been used and stored in
+    detpar = IX_detector_array(detpar);
+end
+if isa(detpar,'IX_detector_array')
+    det = unique_objects_container('baseclass','IX_detector_array');
+    det = det.add(detpar);
+    det = det.replicate_runs(n_instances);
+elseif isa(detpar,'unique_objects_container') || isa(detpar,'unique_references_container')
+    if detpar.n_objects == 1
+        det = detpar.replicate_runs(n_instances);
+    elseif ~isempty(n_instances) && detpar.n_objects == n_instances
+        det = detpar;
+    else
+        error('HORACE:horace_binfile_interface:invalid_argument', ...
+            ['Number of IX_detector_array objects %d provided in ' ...
+            'the input container is not consistent with number of objects, ' ...
+            'requested to return (%d).\n' ...
+            'Input should be either 1 or equal to the number of requested'], ...
+            detpar.n_objects,n_instances)
+    end
+else
+    error('HORACE:horace_binfile_interface:invalid_argument', ...
+        ['Unrecognized input class: "%s" provided to the conversion method\n' ...
+        'Allowed inputs are: "struc", "ID_detector_array", ' ...
+        '"unique_objects_container" or "unique_references_container" '],...
+        class(detpar))
+end
+end
```

Comment:
the above code deals with detpar as either unque-objects_container or uniqu_references_container and returns a unique-objects_container. I would expect returning  references container. In either case can you document the logic

Reply:
Your suggestion would work, and I thought about it. 

But decided, that it is safe to have `unique_obj_container` here. You have `unique_references_container` in `Experiment` and asignment will check if you provide correct value and distribute it within references. 

The routine here just processes old data and check them for uniqness. It is easier to test it this way without addressing global storage.

Due to its global nature, `unique_references_contaienr` should be exposed outside of `Experiment` as little as possible.